src/python/platforms/fuchsia/device.py

```diff
@@ -16,6 +16,119 @@
+                           'qemu-system-x86_64')
+  os.chmod(qemu_path, 0o550)
+  kernel_path = os.path.join(resources_path, 'target', 'x64', 'qemu-kernel.bin')
+  os.chmod(kernel_path, 0o644)
+  authorized_keys_path = os.path.join(resources_path, '.ssh', 'authorized_keys')
+  pkey_path = os.path.join(resources_path, '.ssh', 'pkey')
+  os.chmod(pkey_path, 0o400)
+  sharefiles_path = os.path.join(resources_path, 'qemu-for-fuchsia', 'share',
+                                 'qemu')
+
+  # The FVM is minimally sized to begin with; extend it to make room for
+  # ephemeral packages etc.
+  drive_path = os.path.join(resources_path, 'target', 'x64', 'fvm.blk')
+  os.chmod(drive_path, 0o644)
+  fvm_tool_path = os.path.join(resources_path, 'tools', 'fvm')
+  os.chmod(fvm_tool_path, 0o500)
+  process = new_process.ProcessRunner(fvm_tool_path,
+                                      [drive_path, 'extend', '--length', '1G'])
+  result = process.run_and_wait()
+  if result.return_code or result.timed_out:
+    raise errors.FuchsiaSdkError('Failed to extend FVM: ' + result.output)
+
+  # Need to bake keys into ZBI so we can SSH into it.  See:
+  # fuchsia.googlesource.com/fuchsia/+/refs/heads/master/sdk/docs/ssh.md
+  zbi_tool = os.path.join(resources_path, 'tools', 'zbi')
+  os.chmod(zbi_tool, 0o500)
+  fuchsia_zbi = os.path.join(resources_path, 'target', 'x64', 'fuchsia.zbi')
+  initrd_path = os.path.join(resources_path, 'fuchsia-ssh.zbi')
+  process = new_process.ProcessRunner(zbi_tool, [
+      '-o', initrd_path, fuchsia_zbi, '-e',
+      'data/ssh/authorized_keys=' + authorized_keys_path
+  ])
+  result = process.run_and_wait()
+  if result.return_code or result.timed_out:
+    raise errors.FuchsiaSdkError('Failed to add keys to Fuchsia ZBI: ' +
+                                 result.output)
+  os.chmod(initrd_path, 0o644)
+
+  # TODO(flowerhack): Add a mechanism for choosing portnum dynamically.
+  portnum = '56339'
+
+  # yapf: disable
+  qemu_command = [
+      qemu_path,
+      '-m', '2048',
+      '-nographic',
+      '-kernel', kernel_path,
+      '-initrd', initrd_path,
+      '-smp', '4',
+      '-drive', 'file=' + drive_path + ',format=raw,if=none,id=blobstore',
+      '-device', 'virtio-blk-pci,drive=blobstore',
+      '-monitor', 'none',
+      '-append', '"kernel.serial=legacy TERM=dumb"',
+      '-machine', 'q35',
+      '-enable-kvm',
+      '-display', 'none',
+      '-cpu', 'host,migratable=no',
+      '-netdev',
+      ('user,id=net0,net=192.168.3.0/24,dhcpstart=192.168.3.9,'
+       'host=192.168.3.2,hostfwd=tcp::') + portnum + '-:22',
+      '-device', 'e1000,netdev=net0,mac=52:54:00:63:5e:7b',
+      '-L', sharefiles_path
+  ]
+  # yapf: enable
+  # Fuzzing jobs that SSH into the QEMU VM need access to these env vars.
+  environment.set_value('FUCHSIA_PKEY_PATH', pkey_path)
+  environment.set_value('FUCHSIA_PORTNUM', portnum)
+  qemu_process = new_process.ProcessRunner(qemu_command[0], qemu_command[1:])
+  qemu_process.run()
```

Comment:
can you please add a TODO here for cleanup logic? 

as is, does this create a new process every run?

Reply:
todo added!